src/sub_dirs.ml

```diff
@@ -0,0 +1,112 @@
+open! Stdune
+
+type 'set t =
+  { sub_dirs : 'set
+  ; data_only : 'set
+  }
+
+let default =
+  let standard_dirs =
+    Predicate_lang.of_pred (function
+      | "" -> false
+      | s -> s.[0] <> '.' && s.[0] <> '_')
+  in
+  { sub_dirs = standard_dirs
+  ; data_only = Predicate_lang.empty
+  }
+
+let make ~sub_dirs ~ignored_sub_dirs ~data_only =
+  let sub_dirs =
+    let ignored_subdirs = Predicate_lang.union ignored_sub_dirs in
```

Comment:
shouldn't ignored_subdirs be merged with data_only rather than sub_dirs? To preserve backward compatibility

Reply:
Good call.